admin/mirage/handlers/target-profiles.js

```diff
@@ -0,0 +1,20 @@
+import { Response } from 'ember-cli-mirage';
+
+function attachTargetProfiles(schema, request) {
+  const organizationId = request.params.id;
+  const params = JSON.parse(request.requestBody);
+  const targetProfilesToAttach = params.data.attributes['target-profiles-to-attach'];
+  targetProfilesToAttach.forEach((targetProfileId) =>
+    schema.targetProfiles.create({ organizationId, name: `Profil ${targetProfileId}` }));
+  return new Response(204);
+}
```

Comment:
@lisequesnel C'est pas vital mais ceci n'attache pas des `target-profiles` à une organisation, j'ai plutôt l'impression que ça crée de nouveaux `target-profiles` dont l'organisation _propriétaire_ est celle demandée. Ce qui "marche" parce que lorsqu'on demande les profils cibles d'une organisation on a récupère ceux qu'elle possède _et_ ceux qui lui sont partagés.

J'imagine que c'est fait en connaissance de cause ; mais est-ce que ce comportement Mirage incohérent avec la "vraie" API ne risque pas de créer une perplexité chez la prochaine personne qui essaiera de s'en servir ? 🤔

Reply:
En fait, Mirage reproduit le comportement que le front s'attend à avoir. Et ce qu'il s'attend à avoir c'est de nouveaux profils cibles associés à son organisation. Le front n'a aucune notion de `target-profile-share`, et donc pour lui c'est purement comme une création de profil cible pour une organisation. C'est un peu perturbant, je suis d'accord, mais Mirage fonctionne comme ça. À moins que tu aies autre chose à proposer ? 🙂

Reply:
Oui je proposerais de respecter le modèle de données de Pix, à savoir qu'il existe des TargetProfiles propriété d'orgas et d'autres qui leur sont partagés. C'est vrai que l'API Pix n'expose pas vraiment ce modèle aujourd'hui puisqu'elle ne fournit aucun moyen de créer un TargetProfile. Mais le modèle existe pourtant bien et il est régulièrement manipulé ainsi par les gens (pour l'instant, chez Pix) qui créent et partagent les TargetProfiles.

D'ailleurs, le code ci-dessus ne correspond pas au comportement attendu de l'API puisqu'il ne permet pas avec Mirage de reproduire le cas où on demande d'attacher un profil cible inexistant — ce qui serait pratique pour vérifier que le message d'erreur s'affiche correctement, etc.

Ma suggestion alternative est de créer explicitement le TargetProfile qu'on souhaite ajouter dans le test, comme c'est fait pour le test `should display organization target profiles`:

```js
const otherOrganization = this.server.create('organization');
this.server.create('target-profile', { name: 'Profil cible du ghetto', otherOrganization });
```

Puis l'ID de ce profile cible peut être utilisé pour l'ajout du partage.